wallets/react/tests/legacy.test.tsx

```diff
@@ -0,0 +1,261 @@
+      wrapper,
+    });
+
+    await expect(result.current.connect('whatever')).rejects.toThrow(
+      'You should add'
+    );
+  });
+
+  test('will be ignored and use first namespace when passing multiple namespaces', async () => {
+    const wrapper = ({ children }: any) => {
+      const list = [legacyProviderImportsToVersionsInterface(legacyProvider)];
+
+      return <Provider providers={list}>{children}</Provider>;
+    };
+
+    const { result } = renderHook(() => useWallets(), {
+      wrapper,
+    });
+
+    const connectResult = await result.current.connect('legacy-garbage', [
+      {
+        namespace: LegacyNamespace.Evm,
+        network: 'eth',
+      },
+      {
+        namespace: LegacyNamespace.Evm,
+        network: 'arb',
+      },
+    ]);
+
+    expect(connectResult).toHaveLength(1);
+    expect(connectResult[0].network).toBe('eth');
+  });
+  test('update internal states correctly', async () => {
+    const wrapper = ({ children }: any) => {
+      const list = [legacyProviderImportsToVersionsInterface(legacyProvider)];
+
+      return (
+        <Provider providers={list} allBlockChains={blockchainsMeta}>
+          {children}
+        </Provider>
+      );
+    };
+
+    const { result } = renderHook(() => useWallets(), {
+      wrapper,
+    });
+
+    const initialState = {
+      installed: true,
+      connecting: false,
+      connected: false,
+      accounts: null,
+      network: null,
+    };
+    expect(result.current.state('legacy-garbage')).toMatchObject(initialState);
+
+    // connecting successfully.
+    await result.current.connect('legacy-garbage', [
+      { namespace: LegacyNamespace.Evm, network: 'ETH' },
+    ]);
+
+    expect(result.current.state('legacy-garbage')).toMatchObject({
+      installed: true,
+      connecting: false,
+      connected: true,
+      accounts: [`ETH:${legacyAddress}`],
+      network: 'ETH',
+    });
+
+    // unsuccessful connection
+    await expect(
+      result.current.connect('legacy-garbage', [
+        { namespace: LegacyNamespace.Evm, network: 'When Airdrop?' },
+      ])
+    ).rejects.toThrowError();
+    // state should be reset
+    expect(result.current.state('legacy-garbage')).toMatchObject(initialState);
+  });
+
+  test.todo("throw error when it's already on connecting", async () => {
+    //
+  });
+
+  test.todo('ensure extra context added to `connect` function', async () => {
+    // make sure correct values passed to `provider connect fn`: { instance, network, meta, namespaces}
+  });
+
+  test.todo(
+    'return multiple result from connect should handle network and accounts correctly ',
+    async () => {
+      // if connect fn returns array, it has own logic.
+    }
+  );
+
+  test.todo('return empty result from connect', async () => {
+    // if connect fn return empty array, state shouldn't be updated.
+  });
+
+  test.todo('suggestAndConnect', async () => {
+    //
+  });
+});
+
+describe('check legacy switching network', () => {
```

Comment:
```suggestion
describe('legacy switch network works properly', () => {
```
I think it would be helpful if you follow a specific convention

Reply:
https://github.com/rango-exchange/rango-client/pull/824/files#r1886718647